src/NuGet.Indexing/NuGetAcronymExpansionProvider.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NuGet.Indexing
+{
+    public class NuGetAcronymExpansionProvider : IAcronymExpansionProvider
+    {
+        public static readonly IAcronymExpansionProvider Instance = new NuGetAcronymExpansionProvider();
+
+        private static readonly Dictionary<string, string[]> Acronyms = new Dictionary<string, string[]>(StringComparer.OrdinalIgnoreCase);
+
+        static NuGetAcronymExpansionProvider()
+        {
+            Acronyms.Add("xaml",  new[] { "xaml" });
```

Comment:
I wouldn't put it in a json blob, I too thought this was a good idea, but it will create a bunch of issues when trying to do A/B testing, and add another connectivity pitfall.

I think deployment is fine, I would make it a resource file that is easy to edit, and add a test to validate that the values are valid (e.g. the xml or json loads fine).

I would make it part of this PR